### ISSUE TITLE
py5 Attribute errors could provide useful suggestions

### ISSUE THREAD

**Issue by hx2A** [py5coding/py5generator#224]
When I use git and mistype a command, it will guess at what I meant.

```bash
$ git stutas
git: 'stutas' is not a git command. See 'git --help'.

The most similar command is
        status
(py5) 21:25 jim@main ~/Projects/ITP/pythonprocessing/py5generator:$ git status
On branch splitsketch
Your branch is up to date with 'origin/splitsketch'.

nothing to commit, working tree clean
```

py5 could do this also. If my code includes `py5.collor(100, 200, 300)`  it currently responds with `AttributeError: py5 has no function or field named collor`. What if it instead said it said `AttributeError: py5 has no function or field named collor. Did you mean to type "color"?`

_Originally posted by @hx2A in https://github.com/py5coding/py5generator/issues/198#issuecomment-1399707515_

**Comment by hx2A** [comment]
Also look at http://norvig.com/spell-correct.html for a simple ago and the Python 3.10 feature for Attribute Errors https://docs.python.org/3.10/whatsnew/3.10.html#attributeerrors

**Comment by hx2A** [comment]
@villares , I now have a working prototype for this. Here are a few example error messages:

```txt
py5 encountered an error in your code:

File "<ipython-input-6-b7441cc777da>", line 2, in setup
    1    def setup():
--> 2        c = py5.colorr(255)
    3        py5.background(c)
    ..................................................
     py5.colorr = # AttributeError
          py5 = <module 'py5' from '/Users/jim/INSTALL/anaconda3/envs/
           py5/lib/python3.8/site-packages/py5/__init__.py'>
    ..................................................

AttributeError: py5 has no function or field named "colorr". Did you mean "color"?
```

```txt
py5 encountered an error in your code:

File "<ipython-input-9-e740f3e4b3eb>", line 2, in draw
    1    def draw():
--> 2        py5.rect(py5.mouse_x, py5.mouse_z, 10, 10)
    ..................................................
     py5.mouse_x = 0
     py5.mouse_z = # AttributeError
          py5 = <module 'py5' from '/Users/jim/INSTALL/anaconda3/envs/
           py5/lib/python3.8/site-packages/py5/__init__.py'>
    ..................................................

AttributeError: py5 has no function or field named "mouse_z". Did you mean "mouse_x" or "mouse_y"?
```

This has been added to the Sketch class. I can easily add it to the others, such as Py5Graphics, Py5Image, Py5Shape, etc. This will be a helpful feature for beginners!

**Comment by hx2A** [comment]
Almost ready to make this PR. It is in good shape, just need to do more testing to verify there are no more needed adjustments. This was harder than I thought it would be.

**Comment by villares** [comment]
If you could ping me next weekend I could try building py5 like you instructed me before...

**Comment by hx2A** [comment]
> If you could ping me next weekend I could try building py5 like you instructed me before...

Sure thing! Do you mean this weekend or next weekend?

I just merged this PR so this feature is now a part of py5.

**Comment by villares** [comment]
I meant this weekend, but it will be a bit busy, so maybe next week or next weekend!

**Comment by hx2A** [comment]
> I meant this weekend, but it will be a bit busy, so maybe next week or next weekend!

OK! I need to get a fix in for #254 before it makes sense for you to test. I believe I have a solution available, I just need time to test it.
